lib/apartment/active_record/schema_migration.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 module ActiveRecord
-  class SchemaMigration < ActiveRecord::Base # :nodoc:
+  class SchemaMigration # :nodoc:
     class << self
       def table_exists?
         connection.table_exists?(table_name)
```

Comment:
I believe this monkey-patch needs to be removed as it breaks rails 7.1 support
This is basically an implementation of `table_exists?` from rails 5.2 and has worked for rails < 7.1 because `SchemaMigration` was inheriting from `ActiveRecord::Base` that was implementing `connection` method, after your PR, this method will start raising error "undefined local variable or method `connection' for ActiveRecord::SchemaMigration:Class"
Current implementation of table_exists is here:
```ruby
    def table_exists?
      @pool.with_connection do |connection|
        connection.data_source_exists?(table_name)
      end
    end
```
it's almost identical - the main difference is that it uses `data_source_exists?` on connection while this monkey patch is forcing it to use `table_exists?`
The difference between these 2 methods is here:
```ruby
      # Checks to see if the data source +name+ exists on the database.
      #
      #   data_source_exists?(:ebooks)
      #
      def data_source_exists?(name)
        query_values(data_source_sql(name), "SCHEMA").any? if name.present?
      rescue NotImplementedError
        data_sources.include?(name.to_s)
      end

      # Checks to see if the table +table_name+ exists on the database.
      #
      #   table_exists?(:developers)
      #
      def table_exists?(table_name)
        query_values(data_source_sql(table_name, type: "BASE TABLE"), "SCHEMA").any? if table_name.present?
      rescue NotImplementedError
        tables.include?(table_name.to_s)
      end
```
so basically just `data_source_sql(name)` VS `data_source_sql(table_name, type: "BASE TABLE")`, the difference that this `type:` part does, is that on table_exists it only looks up partitioned tables and relational tables, and on data_source_exists it also looks up views and materialised views, so this difference is irrelevant for schema migrations.

I believe this monkey-patch needs to be removed, this will 100% cause errors on rails 7.1

Reply:
Do you have a failing test I can add then?

Reply:
Do you have a failing test I can add then?

Reply:
Calling `ActiveRecord::SchemaMigration.table_exists?` should be enough,
In rails 7.1 it should raise `undefined local variable or method 'connection' for ActiveRecord::SchemaMigration:Class (NameError)`
In rails 7.0 it should be perfectly fine and return true

Reply:
It looks like this was added because `table_exists?` was removed from SchemaMigration in rails 6.1, but it looks like it was added back in 7.0. Also, you're looking at `main` in Rails. you need to look at the 7.1 branch. As far as I can tell, this monkey patch is fine, as it is identical to the method in 7.0 and 7.1, and fixes the issue in 6.1. You're right that when 7.2 comes out that this will need to be changed again, but it seems fine now